akka-actor/src/main/scala/akka/io/Dns.scala

```diff
@@ -116,7 +116,11 @@ class DnsExt private[akka] (val system: ExtendedActorSystem, resolverName: Strin
         override def apply(r: String): ActorRef = {
           val settings =
             new Settings(system.settings.config.getConfig("akka.io.dns"), "async-dns")
-          val provider = system.dynamicAccess.getClassFor[DnsProvider](settings.ProviderObjectName).get.newInstance()
+          val provider = system.dynamicAccess
+            .getClassFor[DnsProvider](settings.ProviderObjectName)
+            .get
+            .getDeclaredConstructor()
+            .newInstance()
```

Comment:
(this avoids a deprecation warning when compiling with jdk11 against jdk11 instead of against jdk8)